src/searchengine/nova3/nova2.py

```diff
@@ -37,17 +37,21 @@
 import pathlib
 import sys
 import urllib.parse
+from collections.abc import Iterable, Iterator, Sequence
+from enum import Enum
 from glob import glob
 from multiprocessing import Pool, cpu_count
 from os import path
+from typing import Dict, List, Optional, Set, Tuple, Type
 
-THREADED = True
+THREADED: bool = True
```

Comment:
Everything seems to be clear with function signatures. But what exactly is it supposed to do in such cases? Does it prohibit assigning a different type of value to variable in the future? Otherwise, it looks like an overkill, IMO, because it's obvious w/o annotation that its current type is `bool`.

Reply:
>But what exactly is it supposed to do in such cases? Does it prohibit assigning a different type of value to variable in the future?

Yes. It helps the type checker to not be uncertain (or too relaxed) about the type. Without this annotation, `pyright` will not report an error if the variable is reassigned to a different type. `mypy` will still report this incident regardless.

Reply:
First, they invent a language with dynamic typing, then they invent various tools to protect against it... How illogical this world is! (Nevermind. I'm just grumbling.)

BTW, python's dynamic typing (along with its runtime overhead) isn't going anywhere, is it? (This is just the curiosity of a person who does not have much knowledge of the python world.)

Reply:
>First, they invent a language with dynamic typing, then they invent various tools to protect against it... How illogical this world is! (Nevermind. I'm just grumbling.)

I think people found out dynamic typing isn't maintainable in a large codebase so they invent the annotations.
Not saying our python codebase is large, but the annotations is mainly for the plugin writers to figure out what the helper functions are doing.

>BTW, python's dynamic typing (along with its runtime overhead) isn't going anywhere, is it? (This is just the curiosity of a person who does not have much knowledge of the python world.)

Pure dynamic typing is quite productive when it comes to prototyping or small utility tools so it is unlikely to go away (I think).